### PR TITLE
Api provides functionality for Editors to see and delete any article

### PR DESCRIPTION
**Pull request by arunbhalli**
**PT-Story:**  https://www.pivotaltracker.com/story/show/178264690
### User Story 
``` 
As an Api
to provide editor with proper curation tools
we neeed to show all article and provide delete functionality
``` 
## Changes proposed in this pull request: 
* Make editor index / show test
* change journalist index/ show test
* Add published to show serializer
* add DELETE action for only `role: 'editor'`
* Conditional for index action based on role 

##  What I have learned working on this feature: 
*  how to use method for delete article

### CODE REVIEW

File: app/controllers/api/articles_controller.rb [hunk 1]
```diff
@@ -22,7 +22,13 @@ def index
 
   def show
     article = Article.find(params[:id])
-    if article.published?
+    if request.headers[:source] == 'admin-system'
+      if article_evaluation(article)
+        render json: article, serializer: ArticlesShowSerializer
```
Comment (LucasKnudsen) [inline app/controllers/api/articles_controller.rb:27]: If we do this logic here, we should do it in the Update action as well

File: spec/requests/api/editor/editor_can_see_all_articles_spec.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    it 'is expected to return articles sorted by most recent' do
+      expect(response_json['articles'].first['title']).to eq 'Third Article'
+    end
+
+    it "is expected to have a published status of true" do
+      expect(response_json['articles'].first['published']).to eq true
+      
+    end
+
```
Comment (LucasKnudsen) [inline spec/requests/api/editor/editor_can_see_all_articles_spec.rb]: Formattt

File: app/controllers/api/articles_controller.rb (hunk 1 above)
Comment (lukeperrin10) [reply app/controllers/api/articles_controller.rb:27]: Fixed

File: spec/requests/api/editor/editor_can_see_all_articles_spec.rb (hunk 2 above)
Comment (lukeperrin10) [reply spec/requests/api/editor/editor_can_see_all_articles_spec.rb]: Fixed

**Review by tochman — APPROVED** [review]
Good

**Pull request merged**